yang/goldstone-nexthop.yang

```diff
@@ -0,0 +1,166 @@
+module goldstone-nexthop {
+
+  yang-version "1";
+
+  namespace "http://goldstone.net/yang/goldstone-nexthop";
+  prefix gs-nexthop;
+
+  import goldstone-interfaces { prefix gs-if; }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  organization
+    "GoldStone";
+
+  description
+    "Goldstone Nexthop";
+
+  revision 2022-12-14 {
+    description
+      "Initial revision.";
+  }
+
+  typedef optional-ip-address {
+    type union {
+      type inet:ipv4-address;
+      type string {
+        length "0";
+      }
+    }
+  }
+
+  typedef nexthop-type {
+    type enumeration {
+      enum "interface" {
+        description
+          "Specific interface.";
+      }
+      enum "ip4" {
+        description
+          "IPv4 address.";
+      }
+      enum "ip4-interface" {
+        description
+          "IPv4 address and interface.";
+      }
+      enum "blackhole" {
+        description
+          "Unreachable or prohibited.";
+      }
+    }
+    description
+      "Nexthop types.";
+  }
+
+  typedef blackhole-type {
+    type enumeration {
+      enum "unspec" {
+        description
+          "Generic unreachable.";
+      }
+      enum "null" {
+        description
+          "Null type.";
+      }
+      enum "reject" {
+        description
+          "ICMP unreachable.";
+      }
+      enum "prohibited" {
+        description
+          "ICMP admin-prohibited.";
+      }
+    }
+    default "null";
+    description
+      "Nexthop blackhole types.";
+  }
+
+  grouping nexthop-config {
+    leaf nexthop-type {
+      type nexthop-type;
+      mandatory true;
+      description
+        "The nexthop type.";
+    }
+
+    leaf gateway {
+      type optional-ip-address;
+      description
+        "The nexthop gateway address.";
+    }
+
+    leaf interface {
+      type gs-if:interface-ref;
+      description
+        "The nexthop egress interface.";
+    }
+
+    leaf blackhole-type {
+      type blackhole-type;
+      description
+        "A blackhole sub-type, if the nexthop is a blackhole type.";
+    }
+
+    leaf onlink {
+      type boolean;
+      default "false";
+      description
+        "Nexthop is directly connected.";
+    }
+  }
+
+  grouping nexthop-state {
+    leaf active {
+      type boolean;
+      description
+        "Nexthop is active.";
+    }
+
```

Comment:
What is the difference between `active` and `fib` for the nexthop-state?

Reply:
@ipi-claytonpascoal 
Thank you for your question.
When I looked into OcNOS, there are two attributes, `fib-installed` and `active`.
I intended to represent each attribute as `fib` and `active` in the model.